src/runtime/rpc/rpc_host_port.cpp

```diff
@@ -113,4 +146,50 @@ void host_port::assign_group(const char *name)
     _group_host_port->add_ref();
 }
 
+error_s host_port::resolve_addresses(std::vector<rpc_address> &addresses) const
+{
+    CHECK(addresses.empty(), "");
+
+    switch (type()) {
+    case HOST_TYPE_INVALID:
+        return error_s::make(dsn::ERR_INVALID_STATE, "invalid host_port type: HOST_TYPE_INVALID");
+    case HOST_TYPE_GROUP:
+        return error_s::make(dsn::ERR_INVALID_STATE, "invalid host_port type: HOST_TYPE_GROUP");
+    case HOST_TYPE_IPV4:
+        break;
+    }
+
+    rpc_address rpc_addr;
+    // Transfer hostname like "local" & "192.168.0.1".
```

Comment:
```suggestion
    // Resolve hostname like "localhost:80" or "192.168.0.1:8080".
```
from_string_ipv4 will fail if lack of port field, so add port in comments.